### PR TITLE
Read xml safely

### PR DESCRIPTION
**Pull request by mbojan**
I mentioned this in #10: https://github.com/sckott/oai/issues/10#issuecomment-143072521

When parsing an XML containing illegal characters `libxml` throws an error like

```
PCDATA invalid Char value 25 [9]
```

`read_xml_safely` catches the error, removes the characters, and gives a warning.

See also: http://stackoverflow.com/questions/26938905/perl-xmllibxml-parsing-issue-of-invalid-char

### CODE REVIEW

**Comment by mbojan** [comment]
Hmm... I see that Travis failed. I wonder why because tests and checks passed on my machine.

**Comment by sckott** [comment]
only an older version of R (3.1.3) failed, the current stable and devel passed, looking into why it failed

File: tests/testthat/test-read_xml_safely.R [hunk 1]
```diff
@@ -0,0 +1,18 @@
+context("Safe reading of XMLs")
+
+inputs <- c(
+  "<p>Some paragraph with <b>text in bold</b> which should be OK</p>",
+  "<p>Some paragraph with <b>also in bold</b> with a an illegal character \u0019</p>"
+)
+
+test_that("proper XML is parsed correctly", {
+  expect_equal( read_xml_safely(inputs[1]), xml2::read_xml(inputs[1]) )
```
Comment (sckott) [inline tests/testthat/test-read_xml_safely.R]: @mbojan  The Travis errors occur here and the test at line 15.

Looks like the code being run runs fine, but the error happens in the testthat call. The error message for both is 

> Error in unclass(target) : cannot unclass an external pointer

Perhaps it's having trouble comparing `xml_document` objects - perhaps test for some aspect of the parsed object that is just text

Comment (mbojan) [inline tests/testthat/test-read_xml_safely.R]: OK, thanks. I'll look into it.

**Comment by mbojan** [comment]
Fixed. Tests pass now. Ready to merge.

**Comment by mbojan** [comment]
Well, the tests passed when I connected travis on my fork... I failed now for rdevel. Seems like downloading `covr` failed :(

**Comment by mbojan** [comment]
Meanwhile, I found some bugs. Let's hold on with that PR. 

I am closing this and I will resubmit later.

**Comment by sckott** [comment]
okay, thanks

**Pull request closed without merging**